src/main/java/stream/flarebot/flarebot/util/general/GuildUtils.java

```diff
@@ -0,0 +1,302 @@
+        Role role = guild.getRoles().stream()
+                .filter(r -> r.getName().equalsIgnoreCase(s))
+                .findFirst().orElse(null);
+        if (role != null) return role;
+        try {
+            role = guild.getRoleById(Long.parseLong(s.replaceAll("[^0-9]", "")));
+            if (role != null) return role;
+        } catch (NumberFormatException | NullPointerException ignored) {
+        }
+        if (channel != null) {
+            if (guild.getRolesByName(s, true).isEmpty()) {
+                String closest = null;
+                int distance = LEVENSHTEIN_DISTANCE;
+                for (Role role1 : guild.getRoles().stream().filter(role1 -> FlareBotManager.instance().getGuild(guildId).getSelfAssignRoles()
+                        .contains(role1.getId())).collect(Collectors.toList())) {
+                    int currentDistance = StringUtils.getLevenshteinDistance(role1.getName(), s);
+                    if (currentDistance < distance) {
+                        distance = currentDistance;
+                        closest = role1.getName();
+                    }
+                }
+                MessageUtils.sendErrorMessage("That role does not exist! "
+                        + (closest != null ? "Maybe you mean `" + closest + "`" : ""), channel);
+                return null;
+            } else {
+                return guild.getRolesByName(s, true).get(0);
+            }
+        }
+        return null;
+    }
+    //User
+
+    /**
+     * Gets a {@link User} from a string
+     *
+     * @param s the string to get the user from
+     * @return null if the user wasn't found otherwise a {@link User}
+     */
+    public static User getUser(String s) {
+        return getUser(s, null);
+    }
+
+    /**
+     * Gets a {@link User} from a string
+     *
+     * @param s The string to get the user from
+     * @param guildId The id of the {@link Guild} to get the user from
+     * @return null if the user wasn't found otherwise a {@link User}
+     */
+    public static User getUser(String s, String guildId) {
+        return getUser(s, guildId, false);
+    }
+
+    /**
+     * Gets a {@link User} from a string.
+     *
+     * @param s The string to get the user from
+     * @param forceGet If you want to get the user from discord instead of from a guild
+     * @return null if the user wasn't found otherwise a {@link User}
+     */
+    public static User getUser(String s, boolean forceGet) {
+        return getUser(s, null, forceGet);
+    }
+
+    /**
+     * Gets a {@link User} from a string.
+     *
+     * @param s  The string to get the user from
+     * @param guildId The id of the {@link Guild} to get the user from
+     * @param forceGet If you want to get the user from discord instead of from a guild
+     * @return null if the user wasn't found otherwise a {@link User}
+     */
+    public static User getUser(String s, String guildId, boolean forceGet) {
```

Comment:
We really should make this method more efficient...

Reply:
I'm not doing that one. I've spent enough time working on this just to get it working.